backend/local/backend.go

```diff
@@ -97,6 +119,150 @@ func (b *Local) Configure(c *terraform.ResourceConfig) error {
 	return f(c)
 }
 
+func (b *Local) States() ([]string, string, error) {
+	// If we have a backend handling state, defer to that.
+	if b.Backend != nil {
+		if b, ok := b.Backend.(backend.MultiState); ok {
+			return b.States()
+		} else {
+			return nil, "", ErrEnvNotSupported
+		}
+	}
+
+	// the listing always start with "default"
+	envs := []string{backend.DefaultStateName}
+
+	current, err := b.currentStateName()
```

Comment:
Same as above, "current" state should be a TF core thing and backends should just be told directly what state to load. This will simplify backend implementation a lot and shift that complexity into core but at least its shared.